wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wom/WdlomWomExpression.scala

```diff
@@ -0,0 +1,34 @@
+package wdl.draft3.transforms.wdlom2wom
+
+import cats.syntax.validated._
+import common.validation.ErrorOr.ErrorOr
+import wdl.draft3.transforms.wdlom2wom.ValueConsumer.ops._
+import wdl.model.draft3.elements.ExpressionElement
+import wdl.model.draft3.elements.ExpressionElement.{IdentifierLookup, PrimitiveLiteralExpressionElement}
+import wom.expression.{IoFunctionSet, WomExpression}
+import wom.types.WomType
+import wom.values.{WomFile, WomValue}
+
+final case class WdlomWomExpression(expressionElement: ExpressionElement, inputMap: Map[ConsumedValue, FoundConsumedValue]) extends WomExpression {
+  override def sourceString: String = expressionElement.toString
+
+  override def inputs: Set[String] = inputMap.values.map(_.inputName).toSet
+
+  override def evaluateValue(inputValues: Map[String, WomValue], ioFunctionSet: IoFunctionSet): ErrorOr[WomValue] = expressionElement match {
+    case PrimitiveLiteralExpressionElement(v) => v.validNel
+    case IdentifierLookup(identifier) => inputValues.get(identifier).map(_.validNel).getOrElse(s"No value supplied for variable lookup: $identifier".invalidNel)
+    case _ => ??? // TODO other expression elements
+  }
+
+  override def evaluateType(inputTypes: Map[String, WomType]): ErrorOr[WomType] = expressionElement match {
+    case PrimitiveLiteralExpressionElement(v) => v.womType.validNel
+    case id: IdentifierLookup => inputMap((id: ExpressionElement).consumedValues.head).foundType.validNel
+    case _ => ??? // TODO other expression elements
+  }
+
+  override def evaluateFiles(inputTypes: Map[String, WomValue], ioFunctionSet: IoFunctionSet, coerceTo: WomType): ErrorOr[Set[WomFile]] = {
+    Set.empty[WomFile].validNel
+  }
+}
+
+final case class FoundConsumedValue(inputName: String, foundType: WomType)
```

Comment:
is this `inputName` the same as `ConsumedValue.name`?  If so perhaps there's a way this can be structured so that this info is only stated once

Reply:
Sort of... `ConsumedValue` is unlinked, so it doesn't know how much of an expression like `Int x = a.b.c.d` is the value to lookup, and how much is member accesses on the looked up value (eg is there an object called `a` or a task output from `a` called `a.b`).

The `FoundConsumedValue` is the result of looking through all the generated values and finding one that can satisfy some prefix of the expression lookup, so that `inputName` might be `a.b` if it was a call output, or `a` if it was an object.

Reply:
Maybe `UnlinkedConsumedValue` and `LinkedConsumedValue` would be a good idea for a rename?